### ISSUE TITLE
Model Selection and Training

### ISSUE THREAD

**Issue by ahmetrnn** [ahmetrnn/MachineLearningProject#3]
Your task is to select an appropriate classification model, train it using the training data, and justify your choice based on the dataset characteristics.

Choose a Classification Model
• Goal: Select a model that fits the nature of the HR attrition dataset.
• What to Consider:
• Is the dataset imbalanced? (You should know this from Bjorn’s class distribution)
• Are features mostly numerical or a mix?
• Is interpretability important?
• Suggested options:
• Logistic Regression — simple, interpretable
• Random Forest — handles mixed data well, good for feature importance
• XGBoost or Gradient Boosting — powerful, especially for imbalanced or structured data

• What to use:
• from sklearn.linear_model import LogisticRegression
• from sklearn.ensemble import RandomForestClassifier
• from xgboost import XGBClassifier (optional if installed)

Train the Model
• Goal: Fit the selected model using the training data (X_train, y_train)
• Use:
• .fit(X_train, y_train) — to train the model
• .predict(X_test) — to make predictions on the test set
• Save predictions for evaluation in the next step

Justify Your Model Choice
• Goal: Write a short explanation (2–4 sentences) on why you chose that model.
• What to include:
• Whether your model handles feature types well
• Its ability to handle imbalanced data
• Its interpretability or performance in general

  Tip: Add a markdown cell titled “Model Justification” and explain this briefly but clearly.
Notes:
• Make sure to import the model class at the top
• Define the model with important parameters (e.g., random_state=42, or class_weight='balanced' if classes are imbalanced)
• Keep the training and prediction code clean and modular
• Store your predictions in a variable like y_pred or y_test_pred
